src/02-room-details.js

```diff
@@ -49,7 +76,41 @@ function getRoomByDinosaurName(dinosaurs, rooms, dinosaurName) {}
       "Kit Hopkins Education Wing"
     ]
  */
-function getConnectedRoomNamesById(rooms, id) {}
+function getConnectedRoomNamesById(rooms, id) {
+  const arr = []
+  const ids = []
+  let roomExists = false
+  let connectExists = true
+
+  for (const element of rooms) { //> If target id can be found in the 'connectsTo' property, pushes the name of the room and we know the room must exist
+    if (element.connectsTo.includes(id)) {
+      arr.push(element.name)
+      roomExists = true
+    } 
+    ids.push(element.roomId)
+  }
+
+  // If the attribute 'connectsTo' has an invalid id it should return an error
+  // 1.) Could possibly create an array with all the ids and check if each one is accounted for. If not we reassign connectExists from true to false?
+
+  for (const element of rooms) {
+    for (const subElement of element.connectsTo) {
+      if (!ids.includes(subElement)) {
+        connectExists = false
+      }
+    }
+  }
+
+  if (!connectExists) {
+    return `Room with ID of 'incorrect-id' could not be found.`
```

Comment:
this is considered hardcoded. What if the id was `abc1234`, we would still want to get this error message with the incorrect id dynamically inserted in the string. 

In order to check which ID is incorrect you would have to look for each ID in the `rooms` array and the one that is not found you would return the error message